compiler/rustc_target/src/spec/targets/armv7a_vex_v5_linker_script.ld

```diff
@@ -0,0 +1,58 @@
+OUTPUT_FORMAT("elf32-littlearm")
+
+ENTRY(_start)
+
+__user_ram_start = 0x03800000;
+__user_ram_length = 0x04800000;
+__user_ram_end = __user_ram_start + __user_ram_length;
+
+MEMORY {
+    USER_RAM : ORIGIN = __user_ram_start, LENGTH = __user_ram_length
+}
+
+__stack_length = 0x400000;
+/*
+It's currently unclear why subtracting anything is necessary, but it fixes memory permission errors.
+0x100 is an arbitrary number that works.
+*/
+__heap_end = __user_ram_end - __stack_length - 0x100;
+
+SECTIONS {
+    .text : {
+        __text_start = .;
+        KEEP(*(.code_signature))
```

Comment:
It still really irks me that the code signature is here when it's not text, and I wouldn't be surprised if it irks t-compiler as well. Maybe this is just me being stubborn but I really feel like we should actually resolve this issue, especially since it _has_ worked to have it be in its own section, just with occasional inconsistent issues.

Reply:
I completely agree and i'm not against attempts to fix this, its just out of my area of knowledge to take action on it myself and the previous attempts to do this have failed.